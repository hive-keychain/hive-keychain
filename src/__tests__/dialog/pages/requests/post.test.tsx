import { BackgroundCommand } from '@reference-data/background-message-key.enum';
import { render } from '@testing-library/react';
import React from 'react';
import Post from 'src/dialog/pages/requests/post';
import appMocks from 'src/__tests__/dialog/mocks/app-mocks';
import alButton from 'src/__tests__/utils-for-testing/aria-labels/al-button';
import requestPost from 'src/__tests__/utils-for-testing/data/props/dialog/request-post';
import objects from 'src/__tests__/utils-for-testing/helpers/objects';
import { clickAwait } from 'src/__tests__/utils-for-testing/setups/events';
import { PropsRequestPost } from 'src/__tests__/utils-for-testing/types/props-types';
describe('post tests:\n', () => {
  const { methods, spies } = appMocks;
  const { props } = requestPost;
  methods.config();
  it('Must show Post dialog with title', async () => {
    const clonedProps = objects.clone(props) as PropsRequestPost;
    clonedProps.data.title = 'Greatest Post title of all time!';
    const { asFragment } = render(<Post {...clonedProps} />);
    expect(asFragment()).toMatchSnapshot('Request Post title');
  });
  it('Must show Post dialog without title', async () => {
    const clonedProps = objects.clone(props) as PropsRequestPost;
    delete clonedProps.data.title;
    const { asFragment } = render(<Post {...clonedProps} />);
    expect(asFragment()).toMatchSnapshot('Request Post no title');
  });
  it('Must call window.close when clicking on cancel', async () => {
    render(<Post {...props} />);
    await clickAwait([alButton.dialog.cancel]);
    expect(spies.closeWindow).toBeCalledTimes(1);
  });
  it('Must call onConfirm when clicking on confirm', async () => {
    render(<Post {...{ ...props, onConfirm: spies.onConfirm }} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.onConfirm).toBeCalledTimes(1);
  });
  it('Must call sendMessage when clicking on confirm', async () => {
    render(<Post {...props} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.sendMessage).toBeCalledWith({
      command: BackgroundCommand.ACCEPT_TRANSACTION,
      value: {
        data: props.data,
        tab: props.tab,
        domain: props.domain,
        keep: false,
      },
    });
  });
});
