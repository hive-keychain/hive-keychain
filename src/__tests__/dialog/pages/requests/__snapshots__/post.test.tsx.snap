// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post tests:
 Must show Post dialog with title: Request Post title 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Post
      </h2>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Account
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain.tests
      </div>
      <h3>
        Title
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        Greatest Post title of all time!
      </div>
      <h3>
        Permlink
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        https://hive.com/perm-link-1/
      </div>
      <h3>
        Parent Username
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @theghost1980
      </div>
      <h3>
        Parent Permlink
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        https://hive.com/perm-link/
      </div>
      <h3
        aria-label="clickeable-header"
      >
        Body 
        <span
          class="small"
        >
          (click to show)
        </span>
      </h3>
      <div
        aria-label="collapsible-div"
        class="hide"
      >
        <div
          class="operation_item_content"
        >
          body_stringyfied
        </div>
      </div>
      <h3
        aria-label="clickeable-header"
      >
        JSON Metadata (Click to show)
      </h3>
      <div
        aria-label="collapsible-div"
        class="hide"
      >
        <div
          class="operation_item_content"
        >
          <pre
            role="contentinfo"
          >
            metadata_stringyfied
          </pre>
        </div>
      </div>
    </div>
    <div
      class="operation_footer "
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      >
        <div
          class="checkbox-container align-center"
        >
          <div
            class="checkbox-and-label"
          >
            <div
              class="pretty p-default"
            >
              <input
                aria-checked="false"
                id="checkbox-set-keep-inner-input"
                type="checkbox"
                value=""
              />
              <div
                class="state"
              >
                <label
                  for="checkbox-set-keep-inner-input"
                />
              </div>
            </div>
            <div
              aria-label="checkbox-set-keep"
              class="label"
            >
              Do not prompt again to send post transactions from the @keychain.tests account on domain
            </div>
          </div>
        </div>
      </div>
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`post tests:
 Must show Post dialog without title: Request Post no title 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Post
      </h2>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Account
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain.tests
      </div>
      <h3>
        Permlink
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        https://hive.com/perm-link-1/
      </div>
      <h3>
        Parent Username
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @theghost1980
      </div>
      <h3>
        Parent Permlink
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        https://hive.com/perm-link/
      </div>
      <h3
        aria-label="clickeable-header"
      >
        Body 
        <span
          class="small"
        >
          (click to show)
        </span>
      </h3>
      <div
        aria-label="collapsible-div"
        class="hide"
      >
        <div
          class="operation_item_content"
        >
          body_stringyfied
        </div>
      </div>
      <h3
        aria-label="clickeable-header"
      >
        JSON Metadata (Click to show)
      </h3>
      <div
        aria-label="collapsible-div"
        class="hide"
      >
        <div
          class="operation_item_content"
        >
          <pre
            role="contentinfo"
          >
            metadata_stringyfied
          </pre>
        </div>
      </div>
    </div>
    <div
      class="operation_footer "
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      >
        <div
          class="checkbox-container align-center"
        >
          <div
            class="checkbox-and-label"
          >
            <div
              class="pretty p-default"
            >
              <input
                aria-checked="false"
                id="checkbox-set-keep-inner-input"
                type="checkbox"
                value=""
              />
              <div
                class="state"
              >
                <label
                  for="checkbox-set-keep-inner-input"
                />
              </div>
            </div>
            <div
              aria-label="checkbox-set-keep"
              class="label"
            >
              Do not prompt again to send post transactions from the @keychain.tests account on domain
            </div>
          </div>
        </div>
      </div>
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;
